azure-spring-maven-plugin/src/main/java/com/microsoft/azure/maven/spring/prompt/DefaultPrompter.java

```diff
@@ -0,0 +1,273 @@
+    private static final String PROMOT_MESSAGE_NUM_TEMPLATE = "You have input a wrong number(%s), enter a number for %s again %s: ";
+    private static final String PROMOT_MESSAGE_STRING_TEMPLATE = "You have input a wrong value(%s), please input %s again %s: ";
+    private static final int MAX_ITEMS = 10000;
+
+    // this code is copied from https://stackoverflow.com/questions/13011657/advanced-parsing-of-numeric-ranges-from-string
+    // the author of it is: https://stackoverflow.com/users/433790/ridgerunner
+    private static final Pattern REGEX_COMMA_SEPARATED_INTEGER_RANGES = Pattern.compile("# Validate comma separated integers/integer ranges.\n" +
+            "^             # Anchor to start of string.         \n" +
+            "[0-9]+        # Integer of 1st value (required).   \n" +
+            "(?:           # Range for 1st value (optional).    \n" +
+            "  -           # Dash separates range integer.      \n" +
+            "  [0-9]+      # Range integer of 1st value.        \n" +
+            ")?            # Range for 1st value (optional).    \n" +
+            "(?:           # Zero or more additional values.    \n" +
+            "  ,           # Comma separates additional values. \n" +
+            "  [0-9]+      # Integer of extra value (required). \n" +
+            "  (?:         # Range for extra value (optional).  \n" +
+            "    -         # Dash separates range integer.      \n" +
+            "    [0-9]+    # Range integer of extra value.      \n" +
+            "  )?          # Range for extra value (optional).  \n" +
+            ")*            # Zero or more additional values.    \n" +
+            "$             # Anchor to end of string.           ", Pattern.COMMENTS);
+
+    private static final Pattern REGEX_NEXT_INTEGER_RANGE = Pattern.compile("# extract next integers/integer range value.    \n" +
+            "([0-9]+)      # $1: 1st integer (Base).         \n" +
+            "(?:           # Range for value (optional).     \n" +
+            "  -           # Dash separates range integer.   \n" +
+            "  ([0-9]+)    # $2: 2nd integer (Range)         \n" +
+            ")?            # Range for value (optional). \n" +
+            "(?:,|$)       # End on comma or string end.", Pattern.COMMENTS);
+
+    private BufferedReader reader = new BufferedReader(new InputStreamReader(System.in));
+
+    public String promoteString(String name, String defaultValue, String regex, boolean isRequired) throws IOException {
+        final boolean hasDefaultValue = StringUtils.isNotBlank(defaultValue);
+        final String hintMessage = hasDefaultValue ? (" (" + TextUtils.green(defaultValue) + ")") : "";
+        System.out.printf("Please input the %s%s:", name, hintMessage);
+        System.out.flush();
+
+        while (true) {
+
+            final String input = reader.readLine();
+            if (StringUtils.isBlank(input)) {
```

Comment:
I am not sure whether there will be a corner case is that user want to use an empty value but the default value is not empty?

Reply:
The xml configuration in maven forbids empty string value.

Reply:
add ":" to input empty string